Add clusterroles rbac to obs cluster and clean-up

schwesig opened the pull request:
Add clusterroles rbac to obs cluster and clean-up

- affects obs-cluster
- add clusterroles rbac
- streamline: remove redundant github-client-secret.yaml (follow https://github.com/OCP-on-NERC/nerc-ocp-config/pull/316)
- adjust kustomization.yaml to that
- cleanup: remove oauths-clientsecret-nerc, keycloak.mss not needed  (yet)
- adjust rook-ceph-external-cluster to fit to changes in https://github.com/OCP-on-NERC/nerc-ocp-config/pull/348

fixes https://github.com/nerc-project/operations/issues/308

Signed-off-by: ​/Thor(sten)?/ Schwesig <89909507+schwesig@users.noreply.github.com>

larsks commented on cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml:
```diff
@@ -53,13 +55,6 @@ patches:
-  patch: |
-    - op: replace
-      path: /spec/data/0/remoteRef/key
-      value: nerc/nerc-ocp-obs/openshift-config/oauths-clientsecret-nerc
```
Why are we removing this patch?

schwesig replied:
@larsks I thought they are only for keycloak.mss (only in prod right now), and I assumed we don't want this (yet) for the obs, therefore I streamlined it.
But maybe I am wrong in one or both of these thoughts?

larsks replied:
Thanks, that makes sense.

Would you update the commit message to include this information?

schwesig replied:
thanks, will do.

jbasu01 reviewed (approved):
Looks good

schwesig committed:
Add clusterroles rbac to obs cluster and clean-up

- affects obs-cluster
- add clusterroles rbac
- streamline: remove redundant github-client-secret.yaml (follow OCP-on-NERC#316)
- adjust kustomization.yaml to that
- cleanup: remove oauths-clientsecret-nerc, keycloak.mss not needed
  (yet)
-adjust rook-ceph-external-cluster to fit to match changes in OCP-on-NERC#348

fixes nerc-project/operations#308

Signed-off-by: ​/Thor(sten)?/ Schwesig <89909507+schwesig@users.noreply.github.com>
